### PR TITLE
feat(): add an ascii cat to every log message

### PR DESCRIPTION
**Pull request by BrunnerLivio**
This commit adds 100% more cats to your logs.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features) => **Don't need, it's perfect and I know it**
- [x] Docs have been added / updated (for bug fixes / features) => **Docs are for beginners. Read the source code.**

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?
No cats in the logs :(

## What is the new behavior?
Adds an image of an asciified cat to every log message. The cats are taken from Discord #share-your-cat.

Selection_083

### Features

- 100% more cats in the logs
- Huge log files
- Cats
- No dogs
- The new cats kill all the bugs
- More cats
- Did I mention the log files?
- Cat.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: packages/common/services/logger.service.ts [hunk 1]
```diff
@@ -151,8 +152,37 @@ export class Logger implements LoggerService {
...
+      'https://media.discordapp.net/attachments/679231901109714945/680098182625361951/IMG_20180907_142158.jpg?width=332&height=590',
+      'https://media.discordapp.net/attachments/679231901109714945/679965910127345712/unknown.png?width=787&height=590',
+      'https://media.discordapp.net/attachments/679231901109714945/679392244238909451/IMG_6932.jpeg?width=442&height=590',
+      'https://media.discordapp.net/attachments/679231901109714945/679250502789234708/IMG_20191127_094910337.jpg?width=786&height=590',
+      'https://media.discordapp.net/attachments/679231901109714945/679250226384601098/IMG_20190408_131906409.jpg?width=442&height=590',
+      'https://media.discordapp.net/attachments/679231901109714945/679232485074272256/image0.png?width=465&height=590',
+    ];
+    const randomIndex = Math.round(Math.random() * (cats.length - 1));
+
+    asciify(
```
Comment (BrunnerLivio) [inline packages/common/services/logger.service.ts:177]: Maybe we should instead run a docker container with asciify installed and capture the stdout on every log message, so we do not have a dependency on asciify. What do you think?

File: packages/common/services/logger.service.ts [hunk 2]
```diff
@@ -151,8 +152,37 @@ export class Logger implements LoggerService {
...
+      cats[randomIndex],
+      { fit: 'box', width: 50, height: 50 },
```
Comment (BrunnerLivio) [inline packages/common/services/logger.service.ts:179]: Maybe we can add some machine learning here, so it uses the perfect width/height?

**Comment by BrunnerLivio** [comment]
@kamilmysliwiec the build is failing, can you check [the logs](https://circleci.com/gh/nestjs/nest/19502?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)?

**Comment by jmcdo29** [comment]
I absolutely love this! I'd be worried about making the logs hard to read though. And what about possibly removing it via an option? Maybe something like a `cats: boolean` as a part of the logger options? May be more work than necessary

**Comment by BrunnerLivio** [comment]
@jmcdo29

I want to enforce the cats. I want to enforce hapiness.
Cats are not optional, @jmcdo29

File: packages/common/services/logger.service.ts (hunk 1 above)
Comment (kamilmysliwiec) [reply packages/common/services/logger.service.ts:177]: 🔥

File: packages/common/services/logger.service.ts (hunk 2 above)
Comment (kamilmysliwiec) [reply packages/common/services/logger.service.ts:179]: and, depending on your mood, show a different cat 😍

**Comment by kamilmysliwiec** [comment]
How could I have missed it yesterday 🤯 This should have been released as part of v7 already. Great contribution 🔥

**Comment by bashleigh** [comment]
I DESPERATELY NEED THIS FEATURE FOR MY BUSINESS!!! PLEASE MERGE NOWWWW!!

**Comment by BrunnerLivio** [comment]
@kamilmysliwiec jokes aside, how about publishing it under "april-fools" dist tag? I think that would not cause any issue? ;) Maybe something for next year, since we kinda missed the date now haha

```bash
npm publish @nestjs/core --tag "april-fools"
```

**Comment by shekohex** [comment]
Please I need this feature 😢 

Screen Shot 2020-04-02 at 6 58 27 PM

**Comment by kamilmysliwiec** [comment]
OK let's save it for the next year and publish at midnight 😄  ❤️ 

image

**Pull request closed without merging**